lib/active_merchant/billing/gateways/adyen.rb

```diff
@@ -622,20 +622,34 @@ def add_mpi_data_for_network_tokenization_card(post, payment, options)
 
         post[:mpiData] = {}
         post[:mpiData][:authenticationResponse] = 'Y'
-        post[:mpiData][:cavv] = payment.payment_cryptogram
+        if options[:switch_cryptogram_mapping_nt]
+          post[:mpiData][:tokenAuthenticationVerificationValue] = payment.payment_cryptogram
+        else
+          post[:mpiData][:cavv] = payment.payment_cryptogram
+        end
         post[:mpiData][:directoryResponse] = 'Y'
         post[:mpiData][:eci] = payment.eci || '07'
       end
 
-      def add_recurring_contract(post, options = {})
-        return unless options[:recurring_contract_type]
+      def add_recurring_contract(post, options = {}, payment = nil)
+        return unless options[:recurring_contract_type] || payment.is_a?(NetworkTokenizationCreditCard)
 
         post[:recurring] = {}
-        post[:recurring][:contract] = options[:recurring_contract_type]
-        post[:recurring][:recurringDetailName] = options[:recurring_detail_name] if options[:recurring_detail_name]
-        post[:recurring][:recurringExpiry] = options[:recurring_expiry] if options[:recurring_expiry]
-        post[:recurring][:recurringFrequency] = options[:recurring_frequency] if options[:recurring_frequency]
-        post[:recurring][:tokenService] = options[:token_service] if options[:token_service]
+        if payment.is_a?(NetworkTokenizationCreditCard)
```

Comment:
I would also suggest adding `options[:switch_cryptogram_mapping_nt]` here since this is also part of the overall changes

Reply:
I think that according the behavior on the tests, what you suggests is not needed because these fields are neeeded no matter what the cryptogram mapping is

Reply:
Yes, I understand but they are changes failed last time so we just want to be extra careful which is why we want to use the LD flag in general.